lib/currency_conversion/update_worker.ex

```diff
@@ -27,22 +27,57 @@ defmodule CurrencyConversion.UpdateWorker do
 
     refresh_interval = Keyword.get(opts, :refresh_interval, @default_refresh_interval)
 
-    case refresh(opts) do
-      :ok -> {:ok, opts, refresh_interval}
-      {:error, binary} -> {:stop, {:error, binary}}
+    case refresh_interval do
```

Comment:
It should be enough to replace the old `case` with this one:

```elixir
case {refresh(opts), refresh_interval} do
    {:ok, :manual} -> {:ok, opts}
    {:ok, _} -> {:ok, opts, refresh_interval}
    {{:error, binary}, _} -> {:stop, {:error, binary}}
end
```

Not calling the `refresh` initially should be done when implementing the `seed` config and not with interval change.